engine/configurator/conf_cyaml.h

```diff
@@ -0,0 +1,391 @@
+     */
+    te_bool unit;
+    /**
+     * 'def_val' is taken from YAML key 'default' in object
+     * optional; NULL is default value
+     */
+    char *def_val;
+    /**
+     * 'volat' is taken from YAML key 'volatile' in object
+     * optional; 0 is default value
+     */
+    te_bool volat;
+    /**
+     * 'substitution' is taken from YAML key 'substitution' in object
+     * optional; 0 is default value
+     */
+    te_bool substitution;
+    /**
+     * 'no_parent' is taken from YAML key 'no_parent' in object
+     * optional; 0 is default value
+     */
+    te_bool no_parent_dep;
+
+    /** 'depends' array is taken from YAML key 'depends' in object */
+    depends_entry *depends;
+    /** length of 'depends' array */
+    uint depends_count;
+} object_type;
+
+/** Structure for storing instance */
+typedef struct instance_type {
+    /**
+     * 'if_cond' is taken from YAML key 'if' in instance.
+     * The oid will be ignored iff the resolved value would be FALSE.
+     * CYAML doesn'know about this field.
+     */
+    char *if_cond; /* cyaml doesn't know about this field */
+
+    /** 'oid' is taken from YAML key 'oid' in instance */
+    char *oid;
+    /**
+     * 'value' is taken from YAML key 'value' in instance
+     * optional; NULL is default value
+     */
+    char *value;
+} instance_type;
+
+/**
+ * Structure for storing backup entry
+ * The only one of pointers is not NULL
+ */
+typedef struct backup_entry {
+    /** object */
+    object_type *object;
+    /** instance */
+    instance_type *instance;
+} backup_entry;
+
+/** Structure for storing the whole backup. */
+typedef struct backup_seq {
+    /** 'entries' array is taken from the root YAML sequence */
+    backup_entry *entries;
+    /** length of 'entries' array */
+    uint entries_count;
+} backup_seq;
+
+/**
+ * Structure for storing cond node.
+ * cyaml doesn't know about this structure.
+ */
+typedef struct cond_entry {
+    /**
+     * 'if_cond' is taken from YAML key 'if' in 'cond' node
+     * If the resolved value is TRUE then CS should consider 'then' node.
+     * Otherwise 'else' (if it exists).
+     */
+    char *if_cond;
+    /** 'then_cond' is taken from YAML key 'then' in 'cond' node */
+    struct history_seq *then_cond;
+    /** 'else_cond' is taken from YAML key 'else' in 'cond' node */
+    struct history_seq *else_cond;
+} cond_entry;
+
+/* Structure for storing entry of history.
+ * The only one pointer in not equal to NULL. */
+typedef struct history_entry {
+    /**
+     * 'comment' is taken from YAML key 'comment' in root sequence.
+     * It will be ignored.
+     * CYAML doesn'know about this field.
+     */
+    char *comment;
+
+    /**
+     * 'incl' array is taken from YAML key 'include' in root sequence.
+     * It consists of the names of configuration files that needs to include.
+     * CYAML doesn't know about this field.
+     */
+    char **incl;
+    /** length of 'incl' array */
+    uint incl_count;
```

Comment:
I think that it is defined somewhere in libcyaml because it is taken from the examples of using of cyaml. 
It is definetly defined somewhere in TE(?) because I used it not with libcyaml and there was also no errors. I've changed to `unsigned int` previously after review request. 
This was an answer to your question.

I'll switch to `unsigned int` here too.

Reply:
Fixed.